### ISSUE TITLE
TBlob.get_with_shape error when backforward in complicated netwok.

### ISSUE THREAD

**Issue by CitronMan** [dmlc/mxnet#3144]
**I want to add regularization to my network, but failed. The following is my graph of computation:(bi-rnn)**
def bi_lstmUnroll_reg(numLstmLayer: Int, seqLen: Int, inputSize: Int, numHidden: Int,
                 numEmbed: Int, numLabel: Int, dropout: Float = 0f): Symbol = {
	    val embedWeight = Symbol.Variable("embed_weight")
	    val clsWeight = Symbol.Variable("cls_weight")
	    val clsBias = Symbol.Variable("cls_bias")
	
	    var lastStates = Array[LSTMState]()
	    for (i <- 0 until numLstmLayer) {
	      lastStates = lastStates :+ LSTMState(c = Symbol.Variable(s"l${i}_init_c"),
	                                           h = Symbol.Variable(s"l${i}_init_h"))
	    }
	    var i = 0
	    val forward_paramCells = LSTMParam(i2hWeight = Symbol.Variable(s"l${i}_i2h_weight"),
	                                           i2hBias = Symbol.Variable(s"l${i}_i2h_bias"),
	                                           h2hWeight = Symbol.Variable(s"l${i}_h2h_weight"),
	                                           h2hBias = Symbol.Variable(s"l${i}_h2h_bias"))
	    i = 1
	    val backward_paramCells = LSTMParam(i2hWeight = Symbol.Variable(s"l${i}_i2h_weight"),
	                                           i2hBias = Symbol.Variable(s"l${i}_i2h_bias"),
	                                           h2hWeight = Symbol.Variable(s"l${i}_h2h_weight"),
	                                           h2hBias = Symbol.Variable(s"l${i}_h2h_bias"))
	    
	    assert(lastStates.length == numLstmLayer)
	
	    // embeding layer
	    val data = Symbol.Variable("data")
	    var label = Symbol.Variable("softmax_label")
	    val embed = Symbol.Embedding("embed")(Map("data" -> data, "input_dim" -> inputSize,
	                                           "weight" -> embedWeight, "output_dim" -> numEmbed))
	    val wordvec = Symbol.SliceChannel()(Array(embed),
	      Map("num_outputs" -> seqLen, "squeeze_axis" -> true))

 		var forward_hidden =  Array[Symbol]()
    	var hidden: Symbol = null
	    for (seqIdx <- 0 until seqLen) {
	      	hidden = wordvec.get(seqIdx)
	        val nextState = lstm(numHidden, inData = hidden,
	                             prevState = lastStates(0),
	                             param = forward_paramCells,
	                             seqIdx = seqIdx, layerIdx = 0, dropout = dropout)
	        hidden = nextState.h
	        lastStates(0) = nextState
	        forward_hidden :+= hidden
	    }
	    
	    //backward
	    var backward_hidden =  Array[Symbol]()
	    for (seqIdx <- 0 until seqLen) {
	    	val k = seqLen - seqIdx - 1
	      	hidden = wordvec.get(k)
	        val nextState = lstm(numHidden, inData = hidden,
	                             prevState = lastStates(1),
	                             param = backward_paramCells,
	                             seqIdx = seqIdx, layerIdx = 1, dropout = dropout)
	        hidden = nextState.h
	        lastStates(1) = nextState
	        backward_hidden = hidden +:backward_hidden
	    }
	    var hiddenAll = Array[Symbol]()
	    for (seqIdx <- 0 until seqLen) {
	    	hiddenAll :+= Symbol.Concat()(Array(forward_hidden(seqIdx),backward_hidden(seqIdx)), Map("dim"->1))
 	    }
	    val hiddenConcat = Symbol.Concat()(hiddenAll, Map("dim" -> 0))
    	val pred = Symbol.FullyConnected("pred")(Map("data" -> hiddenConcat, "num_hidden" -> numLabel,
                                            "weight" -> clsWeight, "bias" -> clsBias))
    	label = Symbol.transpose(label) //do  need!
    	label = Symbol.Reshape()(Map("data" -> label, "target_shape" -> "(0,)"))

    	val smce = Symbol.Softmax_cross_entropy(pred, label)
    	Symbol.MakeLoss("lossdd")(Map("data"->smce))

return the error info:
[22:51:19] /home/lxg/workspace-c/BrainMatrix/dmlc-core/include/dmlc/logging.h:235: [22:51:19] /home/lxg/workspace-c/BrainMatrix/mshadow/mshadow/tensor_blob.h:698: Check failed: (this->shape_.Size()) == (shape.Size()) TBlob.get_with_shape: new and old shape do not match total elements
……
Then I debuged the program, find the shape.size==0 in tensor_blob.h, so frustrating.

However,When I train a simple multiple layer perceptron network, it goes well!
	def mlp():Symbol = {
	    val data = Symbol.CreateVariable("data")
	    val weight_1 = Symbol.CreateVariable("weight_1")
	    val weight_2 = Symbol.CreateVariable("weight_2")
	    val weight_3 = Symbol.CreateVariable("weight_3")
	    val label = Symbol.CreateVariable("softmax_label")
 		val fc1 = Symbol.FullyConnected()(Map("data" -> data, "name" -> "fc1", "weight"->weight_1,"no_bias"->true,"num_hidden" -> 128))
 		
	    val act1 = Symbol.Activation()(Map("data" -> fc1, "name" -> "relu1", "act_type" -> "relu"))
	    
	    val fc2 = Symbol.FullyConnected()(Map("data" -> act1, "name" -> "fc2", "num_hidden" -> 64))
 		val fc3 = Symbol.FullyConnected()(Map("data" -> data, "name" -> "fc3", "num_hidden" -> 10))
 		val sm = Symbol.SoftmaxOutput("sm")(Map("data" -> fc3))
 		val smce = Symbol.Softmax_cross_entropy(fc3, label)
 		
 		val loss = Symbol.MakeLoss("makeloss")(Map("data"->(smce+Symbol.sum(weight_1))))
 		loss 
  	}

Any reply is appreciated! Thank you!

**Comment by CitronMan** [comment]
In addition,the two model worked well in forward operation.

**Comment by CitronMan** [comment]
@tqchen

**Comment by CitronMan** [comment]
When I come deal with it today, It turns good by itself! Excuse me.
